### PR TITLE
fully remove the rest of the gil-refs

### PR DESCRIPTION
**Pull request by Icxolu**
This fully removes the rest of the gil-refs API including the cargo feature. `cfg-cfg` made it really easy to find all the remaining usages 🎊

### CODE REVIEW

File: pyo3-macros-backend/src/pyclass.rs [hunk 1]
```diff
@@ -1743,20 +1743,7 @@ fn impl_pytypeinfo(cls: &syn::Ident, attr: &PyClassArgs, ctx: &Ctx) -> TokenStre
...
-        unsafe impl #pyo3_path::type_object::HasPyGilRef for #cls {
-            type AsRefTarget = #pyo3_path::PyCell<Self>;
-        }
-    };
-
-    #[cfg(not(feature = "gil-refs"))]
-    let has_py_gil_ref = TokenStream::new();
-
     quote! {
-        #has_py_gil_ref
```
Comment (LilyFoote) [inline pyo3-macros-backend/src/pyclass.rs:1758]: I take it `TokenStream::new()` was just a placeholder to satisfy the compiler?

File: src/gil.rs [hunk 2]
```diff
@@ -224,19 +207,11 @@ impl GILGuard {
...
         increment_gil_count();
 
-        #[cfg(feature = "gil-refs")]
-        #[allow(deprecated)]
-        let pool = mem::ManuallyDrop::new(GILPool::new());
-
         #[cfg(not(pyo3_disable_reference_pool))]
         if let Some(pool) = Lazy::get(&POOL) {
             pool.update_counts(Python::assume_gil_acquired());
         }
```
Comment (LilyFoote) [inline src/gil.rs:213]: Can we get rid of this too?

**Review by LilyFoote — APPROVED** [review]

File: pyo3-macros-backend/src/pyclass.rs (hunk 1 above)
Comment (davidhewitt) [reply pyo3-macros-backend/src/pyclass.rs:1758]: Yep it expands to no tokens :)

File: src/gil.rs (hunk 2 above)
Comment (davidhewitt) [reply src/gil.rs:213]: Sadly not, the "reference pool" is needed for delayed decrefs and so stays around, it's separate from the "GIL Pool" which the GIL Refs were stored in.

**Review by davidhewitt — APPROVED** [review]
Ace, lovely purging of old code! Next step I guess is for us to get those new method names done again :)

**Comment by davidhewitt** [comment]
Oh actually quick thought - this one might be worth having a note down as a "removed deprecated `gil-refs` feature" entry in the CHANGELOG?

**Comment by Icxolu** [comment]
Thanks both for the review!

> Oh actually quick thought - this one might be worth having a note down as a "removed deprecated `gil-refs` feature" entry in the CHANGELOG?

Sure, I'll add that and then proceed to merge 👍

**Pull request merged**